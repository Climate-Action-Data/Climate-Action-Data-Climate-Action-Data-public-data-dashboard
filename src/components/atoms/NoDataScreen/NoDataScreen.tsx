import { Flex } from '@chakra-ui/react'
import { useTranslation } from 'react-i18next'

interface NoDataScreenProps {
  message?: string
}

export const NoDataScreen = (props: NoDataScreenProps) => {
  const { message } = props
  const { t } = useTranslation(`home`)
  return (
    <Flex color="lightGray.600" justifyContent="center" alignItems="center" w="100%" h="100%">
      {message ?? t(`noData`)}
    </Flex>
  )
}
