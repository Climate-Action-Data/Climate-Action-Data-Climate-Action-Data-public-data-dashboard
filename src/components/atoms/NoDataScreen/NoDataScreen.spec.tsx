import { render } from '@testing-library/react'
import { NoDataScreen } from './NoDataScreen'

const DEFAULT_TEST_MESSAGE = `No data available`

describe(`RetirementDetails`, () => {
  it(`renders correctly with message`, () => {
    const { container } = render(<NoDataScreen message={DEFAULT_TEST_MESSAGE} />)
    expect(container).toMatchSnapshot()
  })

  it(`renders correctly with no message`, () => {
    const { container } = render(<NoDataScreen />)
    expect(container).toMatchSnapshot()
  })
})
