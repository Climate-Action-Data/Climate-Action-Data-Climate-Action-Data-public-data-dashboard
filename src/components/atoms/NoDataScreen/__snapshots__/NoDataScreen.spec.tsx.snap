// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetirementDetails renders correctly with message 1`] = `
<div>
  <div
    class="css-jof02b"
  >
    No data available
  </div>
</div>
`;

exports[`RetirementDetails renders correctly with no message 1`] = `
<div>
  <div
    class="css-jof02b"
  >
    --
  </div>
</div>
`;
